boms/user/pom.xml

```diff
@@ -34,12 +34,19 @@
         <relativePath>../../pom.xml</relativePath>
     </parent>
 
-    <groupId>org.wildfly.bom</groupId>
+    <groupId>org.wildfly.bom.builder</groupId>
     <artifactId>wildfly</artifactId>
 
     <packaging>pom</packaging>
 
-    <name>WildFly BOMs: Parent Aggregator</name>
+    <name>WildFly BOM Builders</name>
+
+    <properties>
+        <!-- the BOMs built's groupId -->
+        <bomGroupId>org.wildfly</bomGroupId>
```

Comment:
It seems we've been inconsistent about the bom groupId between the client and server boms, with the latter using org.wildfly.bom and the clients just org.wildfly. This PR switches them all to org.wildfly.